sdk/core/core/inc/az_curl_slist.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_SLIST_H
+#define AZ_CURL_SLIST_H
+
+#include <az_pair.h>
+#include <az_result.h>
+
+#include <curl/curl.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * Appends an HTTP header to the given CURL list.
+ */
+AZ_NODISCARD az_result
+az_curl_slist_append_header(struct curl_slist ** const pp_list, az_pair const header);
```

Comment:
I somewhat perfer just calling curl_slist_append from the curl PAL, but this is OK I suppose (I haven't gotten to its implementation yet).

Reply:
It has a different signature so we have to wrap it for our callbacks.